src/Razor/src/Microsoft.AspNetCore.Razor.Language/BoundAttributeParameterDescriptor.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public abstract class BoundAttributeParameterDescriptor : IEquatable<BoundAttributeParameterDescriptor>
```

Comment:
Most of the `BoundAttributeParameter` boilerplate is just copy-paste-modify from `BoundAttribute`

Reply:
I know Ryan wanted a sealed class here but I'm of the opinion that having an abstract class here is quite nice for testing purposes. I like it as is but calling it out to call it out.

Reply:
Oh yeah forgot to mention that. I tried going that route and it ended up being a lot more churn than I thought. Decided it wasn't worth it.

Reply:
Explain? Like. What if a bunch of get/set properties it just had a public constructor?

Reply:
I'm ok if you guys feel like aligning this with the current design for w/e reason, but when we revisit this area next in a major I want to rethink a lot of these things.